core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -272,4 +273,24 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext {
       sc.stop()
     }
   }
+
+  test("calling multiple sc.stop() must not throw uncaught exception(50) from sparkenv") {
+    var threwNoOrOnlyExceptedException = true
+    try {
+      sc = new SparkContext(new SparkConf().setAppName("test").setMaster("local"))
+      val cnt = sc.parallelize(1 to 4).count()
+      sc.cancelAllJobs()
+      sc.stop()
+      // call stop second time
+      sc.stop()
+    } catch {
+      case e: ServerStateException =>
+        // assert(!e.getMessage.contains("Server is already stopped"))
+        threwNoOrOnlyExceptedException = false
+      case NonFatal(e) =>
+        threwNoOrOnlyExceptedException = true
```

Comment:
what exception are you expecting here?  I don't see any exception when running this test locally.

If you aren't expecting any exception, then you could just have the body of the `try` block in the test (any exception will fail the test).

If there is some "OK" exception, we should probably check more directly for that exception, as this will allow almost any exception to get thrown.